sys/kern/vfs.c

```diff
@@ -341,6 +341,29 @@ int vfs_lookup(const char *path, vnode_t **vp) {
   return error;
 }
 
+int vnr_create(const char *path, vnode_t **dvp, componentname_t *cn) {
+  vnrstate_t vs;
+  vnrstate_init(&vs, VNR_CREATE, path);
+  int error = vfs_nameresolve(&vs);
+  if (error)
+    return error;
+
+  if (vs.vs_vp != NULL) {
+    if (vs.vs_vp != vs.vs_dvp)
+      vnode_put(vs.vs_dvp);
+    else
+      vnode_drop(vs.vs_dvp);
+
+    vnode_drop(vs.vs_vp);
+    return EEXIST;
```

Comment:
Should not the code be returned by `vfs_nameresolve`?
Seems like `vfs_nameresolve` implementation details are leaking here.

Reply:
I didn't put this code in `vfs_nameresolve`, since this case is only handled during `VNR_CREATE` operation. The question is how should vnr operation specific functions look like. Should it contain any extra logic or just be wrappers of `vfs_nameresolve` to simplify calls?

Reply:
Let's postpone the decision. We need to develop `vfs_nameresolve` that handle more use cases. Then we'll reevaluate `vfs_namecreate`.